CRM/Multicompanyaccounting/Hook/BuildForm/FinancialAccount.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+class CRM_Multicompanyaccounting_Hook_BuildForm_FinancialAccount {
+
+  private $form;
+
+  public function __construct($form) {
+    $this->form = $form;
+  }
+
+  public function run() {
+    $this->editOwnerField();
+  }
+
+  /**
+   * Restrict the Owner field
+   * for the finiancial account form,
+   * only legal entity
+   *
+   * @return void
+   * @throws CRM_Core_Exception
+   */
+  private function editOwnerField() {
+    $element = $this->form->getElement('contact_id');
+    $element->setAttribute('data-api-entity', 'Company');
+    $element->setAttribute('data-api-params', json_encode([
+      'search_field' => 'contact_id.legal_name',
```

Comment:
For Contacts of Organisation type, `legal_name` might be empty, but `organization_name` will always be set since it is required for Organisations, so you should use it instead for both `search_field` and `label_field`